server/index.tsx

```diff
@@ -51,13 +57,15 @@ app.use('^/$', async (req, res, _next) => {
   }
 });
 
-app.post(`${APIPrefix}/generate`, async (req, res) => {
+app.post(`${APIPrefix}/generate`, async (req: PreviewHandlerRequest, res) => {
   const rhIdentity = req.headers['x-rh-identity'] as string;
+  const orientationOption = processOrientationOption(req);
 
   if (!rhIdentity) {
     return res.status(401).send('Unauthorized access not allowed');
   }
-  const template: ServiceNames = req.body.template;
+
+  const template: ServiceNames = req.query.template as ServiceNames;
```

Comment:
The post variant of the endpoint has to get all the data from the request body, not query params. I know it's strange and we will definitely streamline it later, but this is how it works now.

Reply:
Hmm is there any particular reason for that? I don't have a problem changing it back of course, just want to understand!

Reply:
There is a legacy reason from when we took over the code. AS I said we will change it in the future (soon-ish) but it has to be like this for now

Reply:
Ah, got it! Thanks for clarifying my friend. I'll have it up post demo 😎